lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c

```diff
@@ -1211,36 +1211,37 @@ int ngap_amf_nas_pdusession_resource_setup_stream(
         &(session_item->PDU_Session_Resource_Setup_Request_Transfer);
 
     /*tx_out*/
-    Ngap_PDUSessionResourceSetupRequestTransfer_t*
-        pduSessionResourceSetupRequestTransferIEs =
-            (Ngap_PDUSessionResourceSetupRequestTransfer_t*)calloc(
-                1, sizeof(Ngap_PDUSessionResourceSetupRequestTransfer_t));
+    Ngap_PDUSessionResourceSetupRequestTransfer_t
+        pduSessionResourceSetupRequestTransferIEs = {0};
 
     ngap_fill_pdu_session_resource_setup_request_transfer(
         amf_pdu_ses_setup_transfer_req,
-        pduSessionResourceSetupRequestTransferIEs);
+        &pduSessionResourceSetupRequestTransferIEs);
     uint32_t buffer_size = 1024;
     char* buffer = (char*)calloc(1, buffer_size);
 
     asn_enc_rval_t er = aper_encode_to_buffer(
         &asn_DEF_Ngap_PDUSessionResourceSetupRequestTransfer, NULL,
-        pduSessionResourceSetupRequestTransferIEs, buffer, buffer_size);
+        &pduSessionResourceSetupRequestTransferIEs, buffer, buffer_size);
 
     if (er.encoded <= 0) {
       OAILOG_ERROR(LOG_NGAP, "PDU Session Resource Request IE encode error \n");
       OAILOG_FUNC_RETURN(LOG_NGAP, RETURNerror);
     }
 
     asn_fprint(stderr, &asn_DEF_Ngap_PDUSessionResourceSetupRequestTransfer,
-               pduSessionResourceSetupRequestTransferIEs);
+               &pduSessionResourceSetupRequestTransferIEs);
 
-    bstring transfer = blk2bstr(buffer, er.encoded);
+    /* Aligned PER encoder of any ASN.1 type. aper_encode function returns the
+    number of encoded bits in the .encoded field of the return value. Use the
+    following formula to convert to bytes: bytes = ((.encoded + 7) / 8) */
+    bstring transfer = blk2bstr(buffer, ((er.encoded + 7) >> 3));
```

Comment:
Do we need similar changes at https://github.com/magma/magma/blob/master/lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c#L700 as we encode same IE

Reply:
Comment addressed.